File: 506 - Relative Ranks.java
import java.util.*;

class Solution {
    public String[] findRelativeRanks(int[] nums) {
        ArrayList<Integer> temp = new ArrayList<>();
        for (int i: nums) {
            temp.add(i);
        }
        temp.sort(Collections.reverseOrder());
        String[] order = new String[nums.length];
        int rank = 0;
        String[] medals = {"0", "Gold Medal", "Silver Medal", "Bronze Medal"};
        int j = 0;
        for (int i: nums) {
            rank = temp.indexOf(i) + 1;
            if (rank <= 3) {
                order[j++] = medals[rank];
                continue;
            }
            order[j++] = Integer.toString(rank);
        }
        return order;
    }
}
class Mycomparator implements Comparator<Integer> {
    @Override
    public int compare(Integer a, Integer b) {
        return a > b ? -1 : 1;
    }
}

File: 506 - Relative Ranks.py
class Solution:
    def findRelativeRanks(self, nums: List[int]) -> List[str]:
        temp = nums[::]
        temp.sort()
        temp = temp[::-1]
        order = []
        medals = [ 0, 'Gold Medal', 'Silver Medal', 'Bronze Medal']
        for score in nums:
            rank = temp.index(score) + 1
            if rank in range(1,4):
                rank = medals[rank]
            rank = str(rank)
            order.append(rank)
        return order
